opm/simulators/flow/DamarisWriter.hpp

```diff
@@ -259,15 +269,81 @@ class DamarisWriter : public EclGenericWriter<GetPropType<TypeTag, Properties::G
                 this->damarisUpdate_ = false; 
             }
 
-            if (this->damarisOutputModule_->getPRESSURE_ptr() != nullptr) 
+            /*if (this->damarisOutputModule_->getPRESSURE_ptr() != nullptr) 
             {
                 dam_err_ = DamarisOutput::setPosition("PRESSURE", rank_,
                                                       this->elements_rank_offsets_[rank_]);
                 dam_err_ = DamarisOutput::write("PRESSURE", rank_,
                                                 this->damarisOutputModule_->getPRESSURE_ptr());
 
                 dam_err_ =  DamarisOutput::endIteration(rank_);
+            }*/
+            
+            
+            // Call damaris_set_position() for all available variables
+            // There is an assumption that all variables are the same size, with the same offset.
+            // see initDamarisTemplateXmlFile.cpp for the Damaris XML descriptions.
+            for ( auto damVar : localCellData ) {
+                // std::map<std::string, data::CellData>
+                const std::string name = damVar.first ;
+                dam_err_ = DamarisOutput::setPosition(name.c_str(), rank_,
```

Comment:
Should this be called uncontitionally for all (possible) variables? Should we not do this inside the `if` below?

Reply:
Probably only needs to be done once (unless the MPI partitioning gets changed)